packages/xml-views-validation/src/validators/elements/use-of-deprecated-class.ts

```diff
@@ -0,0 +1,54 @@
+import { XMLElement } from "@xml-tools/ast";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import { xmlToFQN } from "@ui5-language-assistant/logic-utils";
+import { UseOfDeprecatedClassIssue } from "../../../api";
+
+export function validateUseOfDeprecatedClass(
+  xmlElement: XMLElement,
+  model: UI5SemanticModel
+): UseOfDeprecatedClassIssue[] {
+  const elementTagFqn = xmlToFQN(xmlElement);
+  const ui5Class = model.classes[elementTagFqn];
+  if (ui5Class === undefined) {
+    return [];
+  }
+
+  if (ui5Class.deprecatedInfo !== undefined) {
+    const issues: UseOfDeprecatedClassIssue[] = [];
+    const commonIssueFields: UseOfDeprecatedClassIssue = {
```

Comment:
If you don't define it as `UseOfDeprecatedClassIssue` you won't have to cast to `any` and typescript will force you to add the range property

Reply:
There is a problem with inference of string sub types if I do that:
![image](https://user-images.githubusercontent.com/4233843/80913888-b01fde00-8d50-11ea-8c24-f41877f9f8a5.png)

Reply:
Ok... there is a workaround which looks weird, but I think might be a bit less bad than using undefined (whether cast to `any` or `never`) because the workaround is closed to the definition:
![image](https://user-images.githubusercontent.com/13255544/80914423-0e9a8b80-8d54-11ea-9aa8-4edc95afe895.png)

Reply:
Thanks this is indeed better, but still looks a bit like dark magic 🧙‍♀️

Reply:
Looks like types of properties in object literals are widened if there is no contextual information... I guess this adds the contextual information. I just wish we would have gotten an error if the `as` didn't match the value, but we can't win them all. (It would be possible to get an error if we defined an interface just for this object literal but it looks like overkill).